include/itkANTSRegistration.hxx

```diff
@@ -615,6 +616,41 @@ ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::GenerateData(
   }
   this->SetForwardTransform(forwardTransform);
 
+  if (m_DisplacementFieldSubsamplingFactor > 1)
+  {
+    using TransformType = typename OutputTransformType::TransformType;
+    for (unsigned int i = 0; i < forwardTransform->GetNumberOfTransforms(); ++i)
+    {
+      typename TransformType::Pointer                  transform = forwardTransform->GetNthTransform(i);
+      typename DisplacementFieldTransformType::Pointer displacementFieldTransform =
+        dynamic_cast<DisplacementFieldTransformType *>(transform.GetPointer());
+      if (displacementFieldTransform)
+      {
+        // The transform is a DisplacementFieldTransform
+        displacementFieldTransform->Print(std::cout, 3);
+        const auto displacementField = displacementFieldTransform->GetDisplacementField();
+        m_DisplacementFieldAdaptor->SetTransform(displacementFieldTransform);
+        m_DisplacementFieldAdaptor->SetRequiredOrigin(displacementField->GetOrigin());
+        m_DisplacementFieldAdaptor->SetRequiredDirection(displacementField->GetDirection());
+        auto requiredSize = displacementField->GetLargestPossibleRegion().GetSize();
+        for (unsigned int i = 0; i < requiredSize.GetSizeDimension(); ++i)
+        {
+          requiredSize[i] /= m_DisplacementFieldSubsamplingFactor;
+        }
+        m_DisplacementFieldAdaptor->SetRequiredSize(requiredSize);
+        auto requiredSpacing = displacementField->GetSpacing();
+        for (unsigned int i = 0; i < requiredSpacing.GetVectorDimension(); ++i)
+        {
+          requiredSpacing[i] *= m_DisplacementFieldSubsamplingFactor;
+        }
+        m_DisplacementFieldAdaptor->SetRequiredSpacing(requiredSpacing);
+        m_DisplacementFieldAdaptor->AdaptTransformParameters();
```

Comment:
I don't see `m_DisplacementFieldAdaptor` being used anywhere. Why add all this code to define it?

Reply:
`m_DisplacementFieldAdaptor->AdaptTransformParameters()` operates in-place on the `DisplacementFieldTransform`:

https://github.com/InsightSoftwareConsortium/ITK/blob/e70671dbb059b279b18110a18fc042ef8c62b735/Modules/Registration/Common/include/itkDisplacementFieldTransformParametersAdaptor.hxx#L236-L237